modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -413,6 +458,12 @@ void cv::gimpl::ie::GIEExecutable::run(std::vector<InObj>  &&input_objs,
                           std::back_inserter(context.args),
                           std::bind(&GIEExecutable::packArg, this, _1));
 
+    // NB: Need to store inputs shape to recognize GFrame/GMat
+    ade::util::transform(op.args,
+                         std::back_inserter(context.in_shapes),
+                         [](const cv::GArg& arg) {
+                             return arg.get<cv::gimpl::RcDesc>().shape;
+                         });
```

Comment:
So do you use it really?

Reply:
use it to distinguish `cv::GMat` & `cv::GFrame`